pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentSubscription.java

```diff
@@ -890,6 +901,23 @@ public CompletableFuture<Void> close() {
         }
     }
 
+    protected void retryClose() {
+        ScheduledExecutorService scheduler = topic.getBrokerService().pulsar().getExecutor();
+        scheduler.schedule(() -> {
+            close()
+                    .thenAccept(unused -> {
+                        if (log.isDebugEnabled()) {
+                            log.debug("Close subscription {} finished.", subName);
+                        }
+                    })
+                    .exceptionally(t -> {
+                        log.warn("Close subscription {} failed.", subName, t);
+                        retryClose();
+                        return null;
+                    });
+        }, 20, TimeUnit.SECONDS);
```

Comment:
20 seconds ?
Probably we have to log at WARN level something like

log.warn("Re-scheduling closing of subscription {} in 20 seconds")